### PR TITLE
feat(condo): DOMA-7709 invoice card

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Added invoice card and used `GeneratePaymentLinkService` for generate payment link
изображение

### CODE REVIEW

File: apps/condo/domains/marketplace/components/PageAccess.tsx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { PermissionsRequired } from '@condo/domains/organization/components/OrganizationRequired'
+
+export const InvoiceReadPermissionRequired = ({ children }) => <PermissionsRequired permissionKeys={['canReadInvoices']} children={children} />
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/components/PageAccess.tsx:3]: Please provide React.FC return type here

File: apps/condo/domains/marketplace/components/PageAccess.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+export const InvoiceReadAndManagePermissionRequired = ({ children }) => <PermissionsRequired permissionKeys={['canReadInvoices', 'canManageInvoices']} children={children} />
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/components/PageAccess.tsx:4]: Same as above

File: apps/condo/domains/marketplace/hooks/useCancelStatusModal.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            <Typography.Text type='secondary'>
+                {CancelInvoiceDescription}
+            </Typography.Text>
+        </Modal>
+    )
+}
+
+type CancelModalPropsType = {
+    onCancel?: () => void
+    onButtonClick?: () => void
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/hooks/useCancelStatusModal.tsx:39]: Why this props are not required? What the case of usage CancelModal without callbacks?

File: apps/condo/domains/marketplace/components/Invoice/CreateInvoiceForm.tsx [hunk 4]
```diff
@@ -32,36 +29,21 @@ export const CreateInvoiceForm: React.FC = () => {
...
-                    return getPaymentLinkNotification({ intl, number: createdInvoice.number, url: result.webViewUrl })
-                },
-                onError: (err) => {
-                    console.log(err)
-                },
-                intl,
-            })
+            const paymentLink = await getPaymentLink(createdInvoice.id)
+
+            notification.success(getPaymentLinkNotification({ intl, number: createdInvoice.number, url: paymentLink }))
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/components/Invoice/CreateInvoiceForm.tsx:42]: Payment link can be undefined here? Any checks of that case here?

File: apps/condo/domains/marketplace/hooks/useCancelStatusModal.tsx (hunk 3 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/marketplace/hooks/useCancelStatusModal.tsx:39]: I think only `onButtonClick` is required. When you click on it there is always some side effect (update the form or Invoice). `onCancel` is not used anywhere yet, deleted.

File: apps/condo/domains/marketplace/components/Invoice/CreateInvoiceForm.tsx (hunk 4 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/marketplace/components/Invoice/CreateInvoiceForm.tsx]: Added check in getPaymentLink

File: apps/condo/domains/marketplace/hooks/useInvoicePaymentLink.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                query: GENERATE_PAYMENT_LINK_QUERY,
+                variables: {
+                    data: {
+                        dv: 1,
+                        sender: getClientSideSenderInfo(),
+                        invoices: [{ id: invoiceId }],
+                        callbacks: {
+                            // replace after deciding where to redirect
+                            successUrl: 'https://doma.ai',
+                            failureUrl: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
```
Comment (VKislov) [inline apps/condo/domains/marketplace/hooks/useInvoicePaymentLink.tsx:24]: ＼(٥⁀▽⁀ )／

File: apps/condo/pages/marketplace/invoice/[id]/index.tsx [hunk 6]
```diff
@@ -0,0 +1,386 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+import { Col, Row, RowProps } from 'antd'
+import { get, isEmpty } from 'lodash'
```
Comment (VKislov) [inline apps/condo/pages/marketplace/invoice/[id]/index.tsx:4]:
```
import get from 'lodash/get'
import isEmpty from 'lodash/isEmpty'
```

**Review by VKislov — APPROVED** [review]

File: apps/condo/domains/marketplace/components/Invoice/ResidentPaymentAlert.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            </>
+        )
+    }
+
+    return <Alert
+        type={type}
+        message={message}
+        description={description}
+        showIcon
+    />
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/Invoice/ResidentPaymentAlert.tsx:111]:
```suggestion
    return (
        <Alert
            type={type}
            message={message}
            description={description}
            showIcon
        />
    )
```

File: apps/condo/domains/marketplace/hooks/useInvoicePaymentLink.tsx (hunk 5 above)
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/hooks/useInvoicePaymentLink.tsx:24]: 🥇

File: apps/condo/domains/marketplace/components/Invoice/InvoiceRowsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        },
+    })
+
+    const orderColumns = useInvoiceRowsTableColumns(currencyCode, marketItems)
+
+    const moneyRender = useMemo(() => getMoneyRender(intl, currencyCode), [currencyCode, intl])
+    const totalPrice = rows.filter(row => row.toPay !== '0').reduce((acc, row) => {
+        acc += +row.toPay * row.count
+        return acc
+    }, 0)
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/Invoice/InvoiceRowsTable.tsx:130]:
```suggestion
    const totalPrice = rows.filter(row => row.toPay !== '0').reduce((acc, row) => acc + row.toPay * row.count, 0)
```

File: apps/condo/domains/marketplace/components/Invoice/InvoiceRowsTable.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    const hasMinPrice = rows.find(row => row.isMin)
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/Invoice/InvoiceRowsTable.tsx:131]:
```suggestion
    const hasMinPrice = rows.some(row => row.isMin)
```

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**